ballerina/modules/exporter/exporter.bal

```diff
@@ -0,0 +1,399 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import graphql_schema_registry.parser;
+
+public isolated function export(parser:__Schema schema) returns string|ExportError {
+    string[] sections = [];
+
+    string? schemaSdl = check exportSchemaType(schema);
+    if schemaSdl is string {
+        sections.push(schemaSdl);
+    }
+    string? customDirectivesSdl = check exportDirectives(schema);
+    if customDirectivesSdl is string {
+        sections.push(customDirectivesSdl);
+    }
+    sections.push(check exportTypes(schema));
+
+    return string:'join(DOUBLE_LINE_BREAK, ...sections);
+}
+
+isolated function exportSchemaType(parser:__Schema schema) returns string?|ExportError {
+    if schema.appliedDirectives.length() > 0 {
+        string appliedDirectivesSdl = check exportTypeAppliedDirectives(schema.appliedDirectives);
+
+        map<parser:__Field> fieldMap = {
+            [QUERY_FIELD]: {args: {}, name: QUERY_FIELD, 'type: schema.queryType}
+        };
+        parser:__Type? mutationType = schema.mutationType;
+        if mutationType !is () {
+            fieldMap[MUTATION_FIELD] = {args: {}, name: MUTATION_FIELD, 'type: mutationType};
+        }
+        parser:__Type? subscriptionType = schema.subscriptionType;
+        if subscriptionType !is () {
+            fieldMap[SUBSCRIPTION_FIELD] = {args: {}, name: SUBSCRIPTION_FIELD, 'type: subscriptionType};
+        }
+        string fieldMapSdl = addBraces(addAsBlock(check exportFieldMap(fieldMap, 1)));
+
+        return SCHEMA_TYPE + appliedDirectivesSdl + fieldMapSdl;
+    } else {
+        return ();
+    }
+}
+
+isolated function getKeyValuePair(string key, string value) returns string {
+    return key + COLON + SPACE + value;
```

Comment:
Let's use string templates instead of concatenation in these kind of places. (Do we really need a constant for the SPACE here?
```suggestion
    return string `key${COLON}${SPACE}${value}`;
```